LCM/dsc/engine/ca/CAInfrastructure/CAEngine.c

```diff
@@ -2376,6 +2545,16 @@ MI_Result MI_CALL PerformInventory( _In_ LCMProviderContext *lcmContext,
         return GetCimMIError(r, extendedError,ID_CAINFRA_NEWSESSION_FAILED);
     }
 
+    // Instantiate native resource manager, responsible to load/unload native resource provider.
+    r = NativeResourceManager_New(&providerContext, &(providerContext.nativeResourceManager));
+    if( r != MI_RESULT_OK)
+    {
+        // CleanUpGetCache(outInstances);
```

Comment:
>outInstances [](start = 27, length = 12)

please delete the instance before return

Reply:
Caller should do that.

---
In reply to: [263566200](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/541#discussion_r263566200) [](ancestors = 263566200)